contracts/modules/TransferManager/GeneralTransferManager.sol

```diff
@@ -211,18 +226,23 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
     * @param _toTime is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
     * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
     * @param _canBuyFromSTO is used to know whether the investor is restricted investor or not.
+    * @param _isAccredited is used to differentiate whether the investor is Accredited or not.
     */
-    function _modifyWhitelist(address _investor, uint256 _fromTime, uint256 _toTime, uint256 _expiryTime, bool _canBuyFromSTO) internal {
+    function _modifyWhitelist(address _investor, uint256 _fromTime, uint256 _toTime, uint256 _expiryTime, bool _canBuyFromSTO, bool _isAccredited) internal {
         require(_investor != address(0), "Invalid investor");
-        uint8 canBuyFromSTO = 0;
-        if (_canBuyFromSTO) {
-            canBuyFromSTO = 1;
-        }
-        if (whitelist[_investor].added == uint8(0)) {
-            investors.push(_investor);
+        uint8 added;
+        uint8 canBuyFromSTO;
+        uint8 accredited;
+        IDataStore dataStore = IDataStore(getDataStore());
+        (,,,,added,) = _getValues(_investor, dataStore);
```

Comment:
For gas optimization, we can create a dedicated function for this which just bit shift added and return only added.